test/matgen.hh

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2017-2023, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+
+#ifndef SLATE_GENERATE_MATGEN_HH
+#define SLATE_GENERATE_MATGEN_HH
+
+
+#include "test.hh"
+#include "../matgen/matgen_params.hh"
```

Comment:
@mgates3 Removing everything, and adding just `generate_matrix.hh` does still work. I went ahead and made that change. Quick question: So it is fine to do indirect `#include` for header files? I had been under the impression that it was better to explicitly add the `#include` for a header file.

Reply:
I personally try to think about what's part of the API exported by the header and what's an internal detail.  For example, `matgen_params` is an intrinsic part of the `generate_matrix` function signature, so we can depend on it being provided by `generate_matrix.hh`.  Whereas, `BaseMatrix.hh` including `<algorithm>` is just an side effect of how we implemented the `BaseMatrix` class.  So, downstream files probably shouldn't assume that `BaseMatrix.hh` will provide `algorithm` in case we change the implementation of `BaseMatrix` to no longer need `algorithm`.

But, the determination is always a little subjective.  Ultimately, the goal should be "what will make this code easiest to understand and maintain in the future".